Wikipedia/SDImageCache+PromiseKit.swift

```diff
@@ -20,23 +21,25 @@ extension SDImageCache {
             }
         })
 
-        // make sure promise is rejected if the operation is cancelled
-        self.KVOController.observe(diskQueryOperation,
-                                   keyPath: "isCancelled",
-                                   options: NSKeyValueObservingOptions.New)
-        { _, _, change in
-            let value = change[NSKeyValueChangeNewKey] as! NSNumber
-            if value.boolValue {
-                reject(NSError(domain: NSURLErrorDomain, code: NSURLErrorCancelled, userInfo: nil))
+        if let diskQueryOperation = diskQueryOperation {
+            // make sure promise is rejected if the operation is cancelled
+            self.KVOController.observe(diskQueryOperation,
+                keyPath: "isCancelled",
```

Comment:
The change I made to SDWebImage reports cancellations as errors _when using the `SDWebImageManager`_. This, however, is straight from the cache, which `WMFImageController` uses to async cache queries.  In order for this to report cancellation properly, I _have_ to monitor the operation.  We definitely have the option of factoring this into a category or something and using that approach instead of modifying SDWebImage, but I prefer to not rely on KVO on an `NSOperation` which is _really_ easy to get wrong in custom subclasses.